Debug ff-matrix gha workflow

cderv opened the pull request:
- Update to fix issues
- Update to new version of its initial inspirations smoke-tests workflow

cderv committed:
Use node 20

cderv committed:
No  need for playwright

cderv committed:
Update ff-matrix workflow based on smoke test one

For now still a copy of the smoke test workflow, but we will refactor this to use the smoke test workflow as a base
and then add the additional steps needed for the feature format matrix tests.

cderv committed:
Opt-in on pull requests to run on the current PR

cderv committed:
Add to ff-matrix workflow too

cderv committed:
Is installing node dependencies helping ?

cderv committed:
fixup - only run linux to try fix hanging issue

cderv committed:
fixup - only run linux to try fix hanging issue

cderv committed:
Rename step as this is not smoke test but feature-format-matrix

cderv commented:
This is interesting because running a single doc that fails previously works
- https://github.com/quarto-dev/quarto-cli/actions/runs/15924034685/job/44917132138?pr=12989#step:25:1002

So this means the hanging has to do with multiple calls one after the other somehow

cderv committed:
debug - let's try all crossref tests

cderv committed:
debug - try floats one only

cderv committed:
debug - try two in a row

cderv commented:
So running all crossref test fails: 
````
./run-tests.sh ../dev-docs/feature-format-matrix/qmd-files/crossref/**/*.qmd
````

And running all floats too
````
./run-tests.sh ../dev-docs/feature-format-matrix/qmd-files/crossref/float/**/*.qmd
````

However, running two tests with Mermaid only works
````
./run-tests.sh ../dev-docs/feature-format-matrix/qmd-files/crossref/float/custom/document.qmd ../dev-docs/feature-format-matrix/qmd-files/crossref/float/figure/document.qmd
````

So still haven't found the pattern that creates the issue

cderv committed:
debug - try all files in floats

cderv commented:
And strangely this works 
````
./run-tests.sh \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/caption-location-bottom/document.qmd \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/caption-location-top/document.qmd \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/code-listing/document.qmd \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/code-listing/executable/document.qmd \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/custom/document.qmd \
../dev-docs/feature-format-matrix/qmd-files/crossref/float/figure/document.qmd
````

cderv committed:
debug - I was missing table file

cderv committed:
debug - only run table document

cderv commented:
So it seems `../dev-docs/feature-format-matrix/qmd-files/crossref/float/table/document.qmd` is failing. But I don't see how it could create the hang... 🤔 

Let's see if this is the problem.

cderv committed:
hide table test and try re-running all floats tests

cderv commented:
It is passing ! Let's try full tests. 

But there is something about this test failing that makes the all process hang. Still don't know what

cderv committed:
reactivate all tests

except table one which is currently hidden

cderv committed:
Revert "reactivate all tests"

This reverts commit 8bea247e494db4b981d27ce0f120b92be3f0188c.

cderv committed:
debug - Restore case fail with all floats

cderv committed:
Run tmate for live debug

cderv committed:
debug remove duplicated call

gordonwoodhull commented:
Thanks @cderv for this thorough investigation! This is really weird. 

It seems like maybe we should just remove mermaid from some tests until it's working. I don't think it's essential here.

I think I know what the six Typst failures are; Pandoc output has changed for figures. The reveal and latex failures are probably churn as well.

cderv committed:
Add  a way to skip test on CI based on metadata

````
_quarto:
  tests-on-ci: false
````

`smoke-all.test.ts` should skip tests when this metadata is set to false inside a CI context

This can be useful for mermaid test for example

cderv committed:
Fix revealjs link

Now internal slide links works

cderv committed:
Typst ref now always have #ref( and supplement

Since https://github.com/quarto-dev/quarto-cli/pull/12285

cderv committed:
remove tmate debugging

cderv committed:
Reapply "reactivate all tests"

This reverts commit 4a20493ff48fcf24de051bbd91d9426c4f1197bd.

cderv committed:
Reactivate Windows

cderv committed:
in typst #align does not have new line anymore inserted before box

cderv committed:
Account for Pandoc 3.2.1 new method for ensuring images don’t overflow

There is now a `\pandocbounded` macro used

cderv committed:
fixup - align

cderv committed:
typo in link

cderv committed:
No more figure element with Typst on Table

Since #11010 as we insert the class
https://github.com/quarto-dev/quarto-cli/blob/d761a8b30cc3e939daaa70724b87c479b0e9f498/src/resources/filters/quarto-post/typst.lua#L127-L131

cderv committed:
use test-smokes.yml for ff-matrix

cderv committed:
unused import

[skip-ci]
